doc/decisions/EXPLANATIONS.md

```diff
@@ -83,10 +83,16 @@ This can be:
 This section has links to other decisions with description what the relation is.
 Decisions that give constraints must be listed in "Constraints" above.
 
+> Note:
+> Sometimes the best solution is only understood if the relation between decisions becomes clear.
+
 ## Notes
 
 Here is a full list of off-line discussions, issue trackers, PRs etc. related to this decision.
 Preferable it is linked, but if it is not possible, it can also be in full-text here.
 If particular information is important and not present in any sections above, please quote it here.
 
+Any discarded, incomplete and unexplored ideas/opinions, which are not complete enough to be "Considered Alternatives", can be written here.
+For example, if it is obvious that the idea does not even solve the problem.
```

Comment:
It would also be good to state WHY the idea is not complete enough/does not even solve the problem.

Reply:
I also think it's not necessary to mention every idea that was thrown around. If an idea seems promising, but e.g. after digging through the code turns out to not solve anything, then yes we should mention it. But if an idea is flawed from the beginning, e.g. because of a logical error made by the person suggesting it, then I don't think we need to mention it. That might only confuse people reading the decision later on.

Reply:
> It would also be good to state WHY the idea is not complete enough/does not even solve the problem.

The main decision needs detailed rationale, the considered alternatives at least a short rationale why it was discarded. To also write rationale for notes is imho too much, I clarified in: 679609185c68de4d87245ae0e408d5b72af11b7d

Reply:
I think the issue here is "discarded ideas". For "incomplete" and "unexplored" it there is no need for rationales. But when an idea is discarded, we need to record why. Otherwise, there will only be questions about it in future. Or worse someone will have to spend the time to rediscover why the idea was discarded.

Reply:
I agree, I removed "discarded ideas" as it is too general: 6b6687c63474a1bfbb6128d69480e5fc3a982a27

I expect notes about any discarded ideas (incomplete/unexplored) actually to be quite rare, at least it was rare up to now. To say more about it is IMHO already over-specification. I don't think that our goal is to have a bullet-proof decision meta-specification that specifies every possible content that might end up in a decision.